src/newTest/java/org/opensearch/test/framework/cluster/TestRestClient.java

```diff
@@ -0,0 +1,366 @@
+            puttedResourcesSet.add(path);
+            puttedResourcesList.add(path);
+        }
+
+        return response;
+    }
+
+    public HttpResponse putJson(String path, ToXContentObject body) throws Exception {
+        return putJson(path, Strings.toString(body));
+    }
+
+    public HttpResponse put(String path) throws Exception {
+        HttpPut uriRequest = new HttpPut(getHttpServerUri() + "/" + path);
+        HttpResponse response = executeRequest(uriRequest);
+
+        if (response.getStatusCode() < 400 && trackResources && !puttedResourcesSet.contains(path)) {
+            puttedResourcesSet.add(path);
+            puttedResourcesList.add(path);
+        }
+
+        return response;
+    }
+
+    public HttpResponse delete(String path, Header... headers) throws Exception {
+        return executeRequest(new HttpDelete(getHttpServerUri() + "/" + path), headers);
+    }
+
+    public HttpResponse postJson(String path, String body, Header... headers) throws Exception {
+        HttpPost uriRequest = new HttpPost(getHttpServerUri() + "/" + path);
+        uriRequest.setEntity(new StringEntity(body));
+        return executeRequest(uriRequest, mergeHeaders(CONTENT_TYPE_JSON, headers));
+    }
+
+    public HttpResponse postJson(String path, ToXContentObject body) throws Exception {
+        return postJson(path, Strings.toString(body));
+    }
+
+    public HttpResponse post(String path) throws Exception {
+        HttpPost uriRequest = new HttpPost(getHttpServerUri() + "/" + path);
+        return executeRequest(uriRequest);
+    }
+
+    public HttpResponse patch(String path, String body) throws Exception {
+        HttpPatch uriRequest = new HttpPatch(getHttpServerUri() + "/" + path);
+        uriRequest.setEntity(new StringEntity(body));
+        return executeRequest(uriRequest, CONTENT_TYPE_JSON);
+    }
+
+    public HttpResponse executeRequest(HttpUriRequest uriRequest, Header... requestSpecificHeaders) throws Exception {
+
+        CloseableHttpClient httpClient = null;
+        try {
+
+            httpClient = getHTTPClient();
+
+            if (requestSpecificHeaders != null && requestSpecificHeaders.length > 0) {
+                for (int i = 0; i < requestSpecificHeaders.length; i++) {
+                    Header h = requestSpecificHeaders[i];
+                    uriRequest.addHeader(h);
+                }
+            }
+
+            for (Header header : headers) {
+                uriRequest.addHeader(header);
+            }
+
+            HttpResponse res = new HttpResponse(httpClient.execute(uriRequest));
+            log.debug(res.getBody());
+            return res;
+        } finally {
+
+            if (httpClient != null) {
+                httpClient.close();
+            }
+        }
+    }
+
+    public TestRestClient trackResources() {
+        trackResources = true;
+        return this;
+    }
+
+    private void cleanupResources() {
```

Comment:
It looks like this is attempting to track and remove resources taht were created in the cluster, instead can we just terminal and restart the the cluster to avoid any edge cases or other side effects?

Reply:
This is unused right now, we should remove it.

Reply:
It seems that the method was deleted

Reply:
Looks like `trackResources`, `puttedResourcesSet` and `puttedResourcesList` should be removed since they aren't used

Reply:
done